zebrad/src/components/mempool.rs

```diff
@@ -175,3 +198,22 @@ impl Service<Request> for Mempool {
         }
     }
 }
+
+async fn remove_expired_transactions(
+    storage: &mut storage::Storage,
+    tip_height: zebra_chain::block::Height,
+) -> Result<(), Box<dyn std::error::Error + Send + 'static>> {
+    let ids = storage.tx_ids().iter().map(|&id| id).collect();
+    let transactions = storage.transactions(ids);
+
+    let _ = transactions
+        .iter()
+        .filter(|t| t.transaction.expiry_height().is_some())
+        .map(|t| {
+            if tip_height > t.transaction.expiry_height().unwrap() {
+                storage.remove(&t.id);
+            }
+        });
```

Comment:
Probably this can be done a bit better. It is also untested code.

Reply:
Yea, i assumed we already had code for this but clearly we don't. I think we should return `None` if `expiry_height` field is `0`. The code should be simple but i agree we can work on this in a separated PR in the context of https://github.com/ZcashFoundation/zebra/issues/2387

Reply:
>  I think we should return `None` if `expiry_height` field is `0`. The code should be simple but i agree we can work on this in a separated PR in the context of #2387

Sounds good.

I think handling `0` correctly might be a blocker for merging this PR.
Currently, this PR expires transactions with expiry height `0` as soon as block `1` is committed to the state.
But expiry height `0` should never expire.

Did you want to write the PR to handle `0`?

Reply:
Yea, will do that.

Reply:
Done in #2782.